### PR TITLE
fix: fmt & clippy

### PR DESCRIPTION
**Pull request by u2**
1. fmt
2. clippy
3. remove feature vec_resize_default

### CODE REVIEW

**Comment by mohanson** [comment]
Failed to compile @u2

**Comment by u2** [comment]
https://github.com/rust-lang/rustup.rs/issues/1558

😢

**Comment by mohanson** [comment]
Well, what I want to say is that the `cargo build` failed in your branch

**Comment by u2** [comment]
Yes, now I change to the Mac at home.

File: evm/src/lib.rs [hunk 1]
```diff
@@ -1,5 +1,3 @@
-#![feature(vec_resize_default)]
```
Comment (u2) [inline evm/src/lib.rs]: changed the `resize_default` to `resize` below.

File: evm/src/common.rs [hunk 2]
```diff
@@ -6,9 +6,9 @@ use std::u64;
 #[inline]
 pub fn to_word_size(size: u64) -> u64 {
     if size > u64::MAX - 31 {
-        return u64::MAX >> 5 + 1;
+        return u64::MAX >> (5 + 1);
```
Comment (u2) [inline evm/src/common.rs]: Is it right?

Comment (mohanson) [reply evm/src/common.rs]: No. It's a bug written by me. Use `(u64::MAX >> 5) + 1` instead plz.

File: state/src/trie/codec.rs [hunk 3]
```diff
@@ -71,31 +65,31 @@ impl<H: Hasher> NodeCodec<H> for RLPNodeCodec<H> {
                 } else {
                     Ok(Node::Extension(key, &value))
                 }
-            },
+            }
             Prototype::List(17) => {
                 let mut nodes = [None; 16];
                 for i in 0..nodes.len() {
```
Comment (u2) [inline state/src/trie/codec.rs]: Need help.

**Comment by u2** [comment]
```
$ cargo test                
   Compiling evm v0.1.0 (/Users/u2/cita-vm/evm)                                                                                                                                      
   Compiling cita-vm v0.1.0 (/Users/u2/cita-vm)                                                                                                                                      
    Finished dev [unoptimized + debuginfo] target(s) in 4.13s                                                                                                                        
     Running target/debug/deps/cita_vm-10fa096995b8565a

running 2 tests
test json_tests::vm_test::tests::test_json_tests_parse ... FAILED
test json_tests::general_state_test::tests::test_json_tests_parse ... FAILED

failures:

---- json_tests::vm_test::tests::test_json_tests_parse stdout ----
thread 'json_tests::vm_test::tests::test_json_tests_parse' panicked at 'called `Result::unwrap()` on an `Err` value: Os { code: 2, kind: NotFound, message: "No such file or directory" }', libcore/result.rs:1009:5

---- json_tests::general_state_test::tests::test_json_tests_parse stdout ----
thread 'json_tests::general_state_test::tests::test_json_tests_parse' panicked at 'called `Result::unwrap()` on an `Err` value: Os { code: 2, kind: NotFound, message: "No such file or directory" }', libcore/result.rs:1009:5
note: Run with `RUST_BACKTRACE=1` for a backtrace.


failures:
    json_tests::general_state_test::tests::test_json_tests_parse
    json_tests::vm_test::tests::test_json_tests_parse

test result: FAILED. 0 passed; 2 failed; 0 ignored; 0 measured; 0 filtered out

error: test failed, to rerun pass '--lib'
```

There are some tests failed, no problem?

**Comment by mohanson** [comment]
Maybe `git submodule update --init` needed.

**Comment by mohanson** [comment]
LGTM

**Pull request merged**